added gurkankaymak/hocon to configuration category

gurkankaymak opened the pull request:
Please check if what you want to add to `awesome-go` list meets [quality standards](https://github.com/avelino/awesome-go/blob/master/CONTRIBUTING.md#quality-standard) before sending pull request. Thanks!

**Please provide package links to:**

- github.com repo: https://github.com/gurkankaymak/hocon
- godoc.org: https://godoc.org/github.com/gurkankaymak/hocon
- goreportcard.com: https://goreportcard.com/report/github.com/gurkankaymak/hocon
- coverage service link: https://codecov.io/gh/gurkankaymak/hocon

Very good coverage

Note: that new categories can be added only when there are 3 packages or more.

**Make sure that you've checked the boxes below before you submit PR:**

- [x]  I have added my package in alphabetical order.
- [x]  I have an appropriate description with correct grammar.
- [x]  I know that this package was not listed before.
- [x]  I have added godoc link to the repo and to my pull request.
- [x]  I have added coverage service link to the repo and to my pull request.
- [x]  I have added goreportcard link to the repo and to my pull request.
- [x]  I have read [Contribution guidelines](https://github.com/avelino/awesome-go/blob/master/CONTRIBUTING.md#contribution-guidelines), [maintainers note](https://github.com/avelino/awesome-go/blob/master/CONTRIBUTING.md#maintainers) and [Quality standard](https://github.com/avelino/awesome-go/blob/master/CONTRIBUTING.md#quality-standard).

Thanks for your PR, you're awesome! 👍

avelino commented:
Thank you for contributing with [awesome-go](https://github.com/avelino/awesome-go), we will revise your contribution as soon as possible.

Automation body links content check:
- godoc.org or pkg.go.dev: True
- goreportcard.com: True
- coverage: True

gurkankaymak commented:
1. fixed the pr template and filled the information
2. fixed the doc link
3. updated the `goreportcard` score, running gofmt restored the score to A+ it was dropped because of minor formattings in the latest commit

thanks
